app/controllers/campaigns_controller.rb

```diff
@@ -0,0 +1,40 @@
+class CampaignsController < ApplicationController
+	def index
+		@campaigns = Campaign.all
+	end
+
+	def create
+		@campaign = Campaign.new(campaign_params)
+		if @campaign.save
+			redirect_to @campaign
+		else
+			render 'new'
+		end
+	end
+
+	def new
+		@campaign = Campaign.new
+	end
+
+	def show
+		@campaign = Campaign.find(params[:id])
+	end
+
+	def edit
+		@campaign = Campaign.find(params[:id])
+	end
+
+	def update
+		@campaign = Campaign.find(params[:id])
+		if @campaign.update(campaign_params)
+			redirect_to @campaign
+		else
+			render "edit"
+		end
+	end
+
+	private
```

Comment:
the before_action takes care of that, it runs the method before each of those actions - it just makes the code more clear in a railsy way and let's us easily add or remove it from future actions (plus it's more dry, I think?)